### PR TITLE
move to gcc7

### PR DESCRIPTION
**Pull request by carlodri**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@conda-forge/python-app @minrk I'm not sure if this PR does the job properly, but the goal was to add python.app to the gcc7 channel so that gcc7 recipes can pick it up properly from conda-forge (and not from defaults).

**Comment by minrk** [comment]
Thanks! This disables the main channel builds, which I think might not be desirable, though a new non-gcc7 build is perhaps unlikely before finishing the migration.

**Comment by minrk** [comment]
Maybe it would be better for python.app to be an extra output of the python feedstock itself with an `exact` dependency on the python it's built with?

**Comment by carlodri** [comment]
ok, I see the point! I can add

```yaml
channel_targets:
- conda-forge gcc7 main
```

is this what you meant?

**Comment by carlodri** [comment]
> Maybe it would be better for python.app to be an extra output of the python feedstock itself with an exact dependency on the python it's built with?

AFAIU I think this would be optimal, although I have never dealt with multiple package outputs up to now so I'm not sure where to start from (probably I would need a good example recipe)

**Comment by carlodri** [comment]
@minrk shall we add the `main` channel or are we good with this?

**Comment by minrk** [comment]
I think there still should be two builds per python, one for the main channel and another for gcc7

**Comment by carlodri** [comment]
@minrk ok, but I'm not sure how to add the `main` channel to the build matrix. Can you help me? It seems like this:

```yaml
channel_targets:
  - conda-forge gcc7 
  - conda-forge main
```

does not have the intended effect.

**Comment by minrk** [comment]
The two builds are tied to the compilers. It might be simplest to add a C compiler as a build dependency even though it's unused, instead of using conda_build_config.yaml:

```yaml
requirements:
  build:
    - {{ compiler('c') }}
  host:
  ...
```

**Comment by carlodri** [comment]
@conda-forge/python-app should be good to go!

**Comment by carlodri** [comment]
@cpbotha @ChrisBarker-NOAA can you please merge if you agree with these changes?

File: recipe/meta.yaml
Comment (minrk) [inline recipe/meta.yaml]: One last thing: to bump the version number, since the previous was 200, this should be 1200.

Comment (carlodri) [reply recipe/meta.yaml]: thanks!

**Comment by carlodri** [comment]
thanks for all your help @minrk!

**Review by minrk — APPROVED** [review]
Thanks! This should unblock orange3's rebuild once merged.

**Comment by mariusvniekerk** [comment]
Thanks

**Comment by jakirkham** [comment]
Thanks you :)

**Comment by carlodri** [comment]
Happy to help!

**Pull request merged**
